datafusion/catalog-listing/src/helpers.rs

```diff
@@ -103,6 +103,8 @@ pub fn expr_applicable_for_cols(col_names: &[&str], expr: &Expr) -> bool {
         // - AGGREGATE and WINDOW should not end up in filter conditions, except maybe in some edge cases
         // - Can `Wildcard` be considered as a `Literal`?
         // - ScalarVariable could be `applicable`, but that would require access to the context
+        // TODO: remove the next line after `Expr::Wildcard` is removed
+        #[allow(deprecated)]
```

Comment:
I think if you use `expect(deprecated)` here and below rather than `allow(deprecated)` the compiler will warn / error once we remove Expr::Wildcard (this is a trick I learned from @itsjunetime )

```suggestion
        #[expect(deprecated)]
```

Reply:
Thanks @alamb! I've changed `#[allow(deprecated)]` to `#[expect(deprecated)]` in this PR.

Reply:
I kept the TODO comment in a few places to provide more context, when `Expr::Wildcard` is not directly referred to in the line following `#[expect(deprecated)]`.